File: libs/io/src/main/java/org/kop/libs/io/files/Converter.java
package org.kop.libs.io.files;

import lombok.SneakyThrows;
import lombok.experimental.UtilityClass;
import lombok.val;
import org.apache.commons.io.FileUtils;
import org.jetbrains.annotations.NotNull;

import java.io.File;
import java.util.Base64;

@UtilityClass
public class Converter {

    @SneakyThrows
    public static @NotNull File base64ToFile(String base64, String target) {
        val targetFile = new File(target);
        FileUtils.writeByteArrayToFile(targetFile,
                Base64.getDecoder().decode(base64));
        return targetFile;
    }

    @SneakyThrows
    public static String fileToBase64(File target) {
        return Base64.getEncoder().encodeToString(FileUtils.readFileToByteArray(target));
    }
}

File: libs/io/src/main/kotlin/org/toolkit4j/libs/io/files/Converter.kt
package org.toolkit4j.libs.io.files

import java.io.File
import java.util.*
import lombok.SneakyThrows
import org.apache.commons.io.FileUtils

@SneakyThrows
fun base64ToFile(base64: String?, target: String?): File? {
  val targetFile = target?.let { File(it) }
  FileUtils.writeByteArrayToFile(targetFile, Base64.getDecoder().decode(base64))
  return targetFile
}

@SneakyThrows
fun fileToBase64(target: File?): String {
  return Base64.getEncoder().encodeToString(FileUtils.readFileToByteArray(target))
}
